src/librustdoc/doctest.rs

```diff
@@ -162,6 +170,44 @@ pub fn run(options: Options) -> Result<(), ErrorReported> {
         Some(testing::Options::new().display_output(display_warnings)),
     );
 
+    // Collect and warn about unused externs, but only if we've gotten
+    // reports for each doctest
+    if json_unused_externs {
+        let unused_extern_reports: Vec<_> =
+            std::mem::take(&mut unused_extern_reports.lock().unwrap());
+        if unused_extern_reports.len() == compiling_test_count {
```

Comment:
Why do we have this condition here? Is it for when some of the tests e.g. failed to parse? I think a comment would be good.

Reply:
Yes, in some instances rustc doesn't print the unused externs report. The check is here to catch this. IIRC there is a similar one on the cargo implementation side.